aspnetcore/security/authentication/index.md

```diff
@@ -0,0 +1,107 @@
+
+* After `UseRouting`, so that route information is available for authentication decisions.
+* Before `UseEndpoints`, so that users are authenticated before accessing the endpoints.
+
+## Authentication Concepts
+
+### Authentication scheme
+
+An authentication scheme is:
+
+* A way of authenticating a user.
+* How the app responds if a user isn't authenticated.
+
+In ASP.NET Core, an authentication scheme consists of:
+
+* An authentication handler.
+* Options for configuring that specific instance of the handler.
+
+Authentication schemes have names so that they can be referenced in the app. For example, an authorization policy that specifies which authorization scheme (or schemes) should be used to authenticate the user. When configuring authentication, it's common to specify the default authentication scheme. The default scheme is used unless a resource requests a specific scheme. It's also possible to:
+
+* Specify different default schemes to use for authenticate, challenge, and forbid actions.
+* Combine multiple schemes into one using [policy schemes](xref:security/authentication/policyschemes).
+
+### Authentication handler
+
+An authentication handler:
+
+* Is a type that implements the behavior of a scheme.
+* Is derived from <xref:Microsoft.AspNetCore.Authentication.IAuthenticationHandler>.
+* Has the primary responsibility to authenticate users.
+
+Based on the authentication scheme's configuration and the incoming request context, authentication handlers:
+
+* Construct <xref:Microsoft.AspNetCore.Authentication.AuthenticationTicket> objects representing the user's identity if authentication is successful.
+* Return 'no result' or 'failure' if authentication is unsuccessful.
+* Have methods for challenge and forbid actions for when users attempt to access resources:
+  * They are unauthorized to access.
+  * When they are unauthenticated.
+
+### Challenge
+
+An authentication challenge is the action an authentication scheme takes when an unauthenticated user requests an endpoint that requires authentication. An authentication challenge is issued, for example, on a request to a restricted resource. Authentication challenge examples include:
+
+* Cookie authentication scheme redirecting the user to a login page.
+* The JWT bearer scheme returning a 401 result.
+
+A challenge action should let the user know that they need to be authenticated to access a requested resource.
+
+### Forbid
+
+An authentication handler's forbid action is used when an authenticated user attempts to access a resource they are not permitted to access. A forbid action typically returns a 403 result. In some custom authentication schemes, forbid could result in:
+
+* A redirect to a page where the user can request access to the resource.
+
+For example, the user my request access from an admin. A forbid action should let the user know:
```

Comment:
can, not should. Arguably 401/403 don't let the user know, it's the app response to the status codes that does.

Reply:
Makes sense. I'll adjust this.